hawkbit-core/src/main/java/org/eclipse/hawkbit/repository/DistributionSetFields.java

```diff
@@ -59,35 +63,39 @@
     /**
      * The metadata.
      */
-    METADATA("metadata", "key", "value");
+    METADATA("metadata", new SimpleImmutableEntry<>("key", "value"));
 
     private final String fieldName;
-    private String keyFieldName;
-    private String valueFieldName;
+    private boolean mapField;
+    private Entry<String, String> subEntityMapTuple;
+
+    DistributionSetFields(final String fieldName) {
+        this(fieldName, false, null);
+    }
 
-    private DistributionSetFields(final String fieldName) {
-        this(fieldName, null, null);
+    DistributionSetFields(final String fieldName, final Entry<String, String> subEntityMapTuple) {
```

Comment:
Is there a reason why you don't keep enum constructors `private`?

Reply:
fixed